x/cdp/keeper/interest.go

```diff
@@ -9,19 +9,80 @@ import (
 	"github.com/kava-labs/kava/x/cdp/types"
 )
 
-// CalculateFees returns the fees accumulated since fees were last calculated based on
-// the input amount of outstanding debt (principal) and the number of periods (seconds) that have passed
-func (k Keeper) CalculateFees(ctx sdk.Context, principal sdk.Coin, periods sdk.Int, collateralType string) sdk.Coin {
-	// how fees are calculated:
-	// feesAccumulated = (outstandingDebt * (feeRate^periods)) - outstandingDebt
-	// Note that since we can't do x^y using sdk.Decimal, we are converting to int and using RelativePow
-	feePerSecond := k.getFeeRate(ctx, collateralType)
-	scalar := sdk.NewInt(1000000000000000000)
-	feeRateInt := feePerSecond.Mul(sdk.NewDecFromInt(scalar)).TruncateInt()
-	accumulator := sdk.NewDecFromInt(types.RelativePow(feeRateInt, periods, scalar)).Mul(sdk.SmallestDec())
-	feesAccumulated := (sdk.NewDecFromInt(principal.Amount).Mul(accumulator)).Sub(sdk.NewDecFromInt(principal.Amount))
-	newFees := sdk.NewCoin(principal.Denom, feesAccumulated.TruncateInt())
-	return newFees
+var (
+	scalingFactor  = 1e18
+	secondsPerYear = 31536000
+)
+
+// AccumulateInterest calculates the new interest that has accrued for the input collateral type based on the total amount of principal
+// that has been created with that collateral type and the amount of time that has passed since interest was last accumulated
+func (k Keeper) AccumulateInterest(ctx sdk.Context, ctype string) error {
+	previousAccrualTime, found := k.GetPreviousAccrualTime(ctx, ctype)
+	if !found {
+		k.SetPreviousAccrualTime(ctx, ctype, ctx.BlockTime())
+		return nil
+	}
+
+	timeElapsed := ctx.BlockTime().Unix() - previousAccrualTime.Unix()
+	if timeElapsed == 0 {
+		return nil
+	}
+
+	totalPrincipalPrior := k.GetTotalPrincipal(ctx, ctype, types.DefaultStableDenom)
+	if totalPrincipalPrior.IsZero() || totalPrincipalPrior.IsNegative() {
+		return nil
+	}
+
+	interestFactorPrior, foundInterestFactorPrior := k.GetInterestFactor(ctx, ctype)
+	if !foundInterestFactorPrior {
+		interestFactorPrior = sdk.OneDec()
+		k.SetInterestFactor(ctx, ctype, interestFactorPrior)
```

Comment:
```suggestion
		k.SetInterestFactor(ctx, ctype, sdk.OneDec())
```

No need to set `interestFactorPrior` since we return at the end of this condition

Reply:
80651c3